src/SSVOpenHexagon/Components/CPlayer.cpp

```diff
@@ -121,23 +170,45 @@ void CPlayer::drawDeathEffect(HexagonGame& mHexagonGame)
 
     for(auto i(0u); i < mHexagonGame.getSides(); ++i)
     {
-        const float sAngle{div * 2.f * i};
+        const float sAngle{fieldAngle + angle + div * 2.f * i};
 
-        sf::Vector2f p1{ssvs::getOrbitRad(pos, sAngle - div, radius)};
-        sf::Vector2f p2{ssvs::getOrbitRad(pos, sAngle + div, radius)};
-        sf::Vector2f p3{
-            ssvs::getOrbitRad(pos, sAngle + div, radius + thickness)};
-        sf::Vector2f p4{
-            ssvs::getOrbitRad(pos, sAngle - div, radius + thickness)};
+        sf::Vector2f p1{
+            Utils::getSkewedOrbitRad(_pos, sAngle - div, radius, skew)};
+        sf::Vector2f p2{
+            Utils::getSkewedOrbitRad(_pos, sAngle + div, radius, skew)};
+        sf::Vector2f p3{Utils::getSkewedOrbitRad(
+            _pos, sAngle + div, radius + thickness, skew)};
+        sf::Vector2f p4{Utils::getSkewedOrbitRad(
+            _pos, sAngle - div, radius + thickness, skew)};
 
         mHexagonGame.wallQuads.reserve_more(4);
         mHexagonGame.wallQuads.batch_unsafe_emplace_back(
             colorMain, p1, p2, p3, p4);
     }
 }
 
+void CPlayer::swap(HexagonGame& mHexagonGame, bool mSoundTog)
+{
+    angle += ssvu::pi;
+    mHexagonGame.runLuaFunctionIfExists<void>("onCursorSwap");
+    if(mSoundTog)
+    {
+        mHexagonGame.getAssets().playSound("swap.ogg");
+    }
+}
+
 void CPlayer::update(HexagonGame& mHexagonGame, ssvu::FT mFT)
 {
+    startPos = mHexagonGame.getFieldPos();
+    sf::Vector2f lastPos{pos};
+    float currentSpeed{speed};
+
+    const float lastAngle{angle};
+    const float radius{mHexagonGame.getRadius()};
+    const int movement{mHexagonGame.getInputMovement()};
+
+    angle += ssvu::toRad(currentSpeed * movement * mFT);
```

Comment:
Why was this moved up? Now `currentSpeed` will lag one frame behind, for example when using focus.